packages/nuxt/src/app/plugins/http-client-hints/types.ts

```diff
@@ -0,0 +1,35 @@
+export interface HttpClientHintsInfo {
+  firstRequest: boolean
+  prefersColorSchemeAvailable: boolean
+  prefersReducedMotionAvailable: boolean
+  viewportHeightAvailable: boolean
+  viewportWidthAvailable: boolean
+}
+export interface HttpClientHintsRequest extends HttpClientHintsInfo {
+  prefersColorScheme?: 'dark' | 'light' | 'no-preference'
+  prefersReducedMotion?: 'no-preference' | 'reduce'
+  viewportHeight?: number
+  viewportWidth?: number
+  colorSchemeFromCookie?: string
+  colorSchemeCookie?: string
+}
+
+export interface HttpClientHintsConfiguration {
```

Comment:
should we also add cookie for viewport w/h and reduce motion for browsers not supporting client hints?